docs/howto/code/custom-action-clearsession/clearsession_action_charm.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python3
+# Copyright 2025 root
+# See LICENSE file for licensing details.
+
+"""Django Charm entrypoint."""
+
+import logging
+import typing
+
+import ops
+
+import paas_charm.django
+
+logger = logging.getLogger(__name__)
+
+
+class DjangoHelloWorldCharm(paas_charm.django.Charm):
+    """Django Charm service."""
+
+    def __init__(self, *args: typing.Any) -> None:
+        """Initialize the instance.
+
+        Args:
+            args: passthrough to CharmBase.
+        """
+        super().__init__(*args)
+        self.framework.observe(self.on.clearsession_action,
+                               self._on_clearsession_action)
+
+    def _on_clearsession_action(self, event: ops.ActionEvent) -> None:
+        """Handle the clearsession action.
+        Args:
+            event: the action event object.
+        """
+        if not self.is_ready():
+            event.fail("django-app container is not ready")
+        try:
+            self._container.exec(
+                ["python3", "manage.py", "clearsessions"],
+                service_context="django",
+                combine_stderr=True,
+                working_dir=str(self._workload_config.app_dir),
+            ).wait_output()
+            event.set_results({"result": "session cleared!"})
+        except ops.pebble.ExecError as e:
+            event.fail(str(e.stdout))
```

Comment:
This should be `stderr` shouldn't it? Same for the other charm

```suggestion
            event.fail(str(e.stderr))
```

Reply:
Oh, I just see that for `_on_clearsession_action` you have `combine_stderr=True` in the `exec` call, which means that `e.stdout` would be the correct thing to use (stdout and stderr will be combined into stdout) - sorry for the misleading comment

For `_on_updatelogfile_action` you're not doing an `exec` call, so I think the errors need to be handled differently from what you currently have. I'll double check now